### PR TITLE
[WIP] Refactor/rework search and gather to allow queries across multiple DB.

### LINKED ISSUE #189: on sbt_search and _gather, make sure to check ksize at load time

### LINKED ISSUE #231: reorder sbt gather/search arguments?
I've never been happy with the order of the sbt_search and sbt_gather arguments; currently, they take SBT first, query signature second:

`sourmash sbt_search genbank.sbt.json query.sig`

and my fingers always want the arguments to be the other way around.

In thinking about #229, I'm wondering if we should just change the order like so:

`sourmash sbt_search query.sig genbank.sbt.json`

which would then support multiple SBTs more naturally:

`sourmash sbt_search query.sig genbank.sbt.json foo.sbt.json bar.sbt.json`

any thoughts @luizirber @taylorreiter @brooksph?

Hmm, we could even combine `search` and `sbt_search` by allowing search to take both SBTs and individual signatures. Whaddya think? Too confusing?

### PR DESCRIPTION
**Pull request by ctb**
Implements better order for sbt_search, allows simultaneous search of multiple SBTs and signatures (part of #229), and renames sbt_search to search (#231).

Removes YAML (#213) and cardinality stuff (#212) now that new databases are available.

Also implements `--output-unassigned` which addresses the rest of #229.

Addresses #215 (results -> stdout) by default and implements `--quiet` flag for most commands.

TODO:

- [x] merge in #172, which, whoops, I thought had been merged into master.
- [x] reorder SBT gather arguments & allow for multisearch
- [x] rename sbt_index to index
- [x] add source filename and md5sum to results (#239)
- [x] update and test tutorial!
- [x] check ksize on load of SBT (#189)
- [x] think about & test similarity vs containment vs common hashes in gather & search
- [x] test `--output-unassigned` on gather
- [x] add `--scaled` to search
- [x] test `--scaled` on gather and search
- [x] properly set `max_hash` in `--output-unassigned` for gather
- [x] test CSV output in search, gather

Checklist:

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

Fixes #189, #192, #212, #213, #229, #230, #231, #239.

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review and maybe merge @luizirber @betatim.

File: sourmash_lib/commands.py [hunk 1]
```diff
@@ -528,27 +467,33 @@ def sbt_combine(args):
...
 
     parser = argparse.ArgumentParser()
     parser.add_argument('sbt_name', help='name to save SBT into')
     parser.add_argument('signatures', nargs='+',
                         help='signatures to load into SBT')
-    parser.add_argument('-k', '--ksize', type=int, default=None)
-    parser.add_argument('--traverse-directory', action='store_true')
+    parser.add_argument('-q', '--quiet', action='store_true',
+                        help='suppress non-error output')
+    parser.add_argument('-k', '--ksize', type=int, default=None,
```
Comment (luizirber) [inline sourmash_lib/commands.py]: Use `sourmash_args.add_ksize_arg` here too?

File: sourmash_lib/commands.py [hunk 2]
```diff
@@ -661,6 +697,8 @@ def categorize(args):
     parser.add_argument('sbt_name', help='name of SBT to load')
     parser.add_argument('queries', nargs='+',
                         help='list of signatures to categorize')
+    parser.add_argument('-q', '--quiet', action='store_true',
+                        help='suppress non-error output')
     parser.add_argument('-k', '--ksize', type=int, default=None)
```
Comment (luizirber) [inline sourmash_lib/commands.py]: `add_ksize_arg` again

File: sourmash_lib/commands.py [hunk 3]
```diff
@@ -828,11 +904,11 @@ def format_bp(bp):
         # based either on an explicit --scaled parameter, or on genome
         # cardinality (deprecated)
         if not best_leaf.minhash.max_hash:
-            error('Best hash match in sbt_gather has no cardinality')
+            error('Best hash match in sbt_gather has no max_hash')
             error('Please prepare database of sequences with --scaled')
             sys.exit(-1)
 
-        R_genome = sourmash_lib.MAX_HASH / float(best_leaf.minhash.max_hash)
+        R_genome = best_leaf.minhash.scaled
```
Comment (luizirber) [inline sourmash_lib/commands.py]: :clap:

**Comment by luizirber** [comment]
This LGTM, the `add_ksize_arg` comments can be dealed with in other PRs (this one is already pretty big)

**Comment by luizirber** [comment]
anything else you want to change @ctb ?

**Comment by ctb** [comment]
No - I will push the merge button! Thank you for the review!

(whew, what a monster.)

**Pull request merged**